src/benchmark.cc

```diff
@@ -684,505 +480,497 @@ void Benchmark::AddRange(std::vector<int>* dst, int lo, int hi, int mult) {
+  if (reporter->ReportContext(context)) {
+    for (const auto& bench_instance : benchmarks) {
+      RunBenchmark(bench_instance, reporter);
+    }
+  }
 }
 
-void FindMatchingBenchmarkNames(const std::string& spec,
-                                std::vector<std::string>* benchmark_names) {
-  if (spec.empty()) return;
+} // end namespace internal
+
 
-  std::vector<internal::Benchmark::Instance> benchmarks;
-  BenchmarkFamilies::GetInstance()->FindBenchmarks(spec, &benchmarks);
-  std::transform(benchmarks.begin(), benchmarks.end(), benchmark_names->begin(),
-                 [](const internal::Benchmark::Instance& b) { return b.name; });
+void RunSpecifiedBenchmarks(const BenchmarkReporter* reporter) {
+  std::string spec = FLAGS_benchmark_filter;
+  if (spec.empty()) {
+    // Nothing to do
+  } else {
+    if (spec.empty() || spec == "all")
+        spec = ".";  // Regexp that matches all benchmarks
+    internal::ConsoleReporter default_reporter;
+    internal::RunMatchingBenchmarks(spec, reporter ? reporter : &default_reporter);
+    std::exit(0);
+  }
 }
 
-}  // end namespace internal
+void SetLabel(const char* label) {
+  CHECK(running_benchmark);
+  MutexLock l(GetBenchmarkLock());
+  report_label = label;
+}
 
-void RunSpecifiedBenchmarks(const BenchmarkReporter* reporter /*= nullptr*/) {
-  std::string spec = FLAGS_benchmark_filter;
-  if (spec.empty() || spec == "all")
-    spec = ".";  // Regexp that matches all benchmarks
-  internal::ConsoleReporter default_reporter;
-  internal::RunMatchingBenchmarks(
-      spec, reporter == nullptr ? &default_reporter : reporter);
+void UseRealTime() {
+  MutexLock l(GetBenchmarkLock());
+  use_real_time = true;
 }
 
+namespace internal {
+
+void StopBenchmarkTiming() {
+  // Add in time accumulated so far
+  CHECK(running_benchmark);
+  timer_manager->StopTimer();
+}
+
+void StartBenchmarkTiming() {
+  CHECK(running_benchmark);
+  timer_manager->StartTimer();
+}
+
+void PrintUsageAndExit() {
+  fprintf(stdout,
+          "benchmark"
+          " [--benchmark_filter=<regex>]\n"
+          "          [--benchmark_iterations=<iterations>]\n"
+          "          [--benchmark_min_iters=<iterations>]\n"
+          "          [--benchmark_max_iters=<iterations>]\n"
+          "          [--benchmark_min_time=<min_time>]\n"
+          "          [--benchmark_repetitions=<num_repetitions>]\n"
+          "          [--benchmark_use_picoseconds={true|false}]\n"
+          "          [--color_print={true|false}]\n"
+          "          [--v=<verbosity>]\n");
+  exit(0);
+}
+
+void ParseCommandLineFlags(int* argc, const char** argv) {
+  using namespace benchmark;
+  for (int i = 1; i < *argc; ++i) {
+    if (
+        ParseStringFlag(argv[i], "benchmark_filter",
+                        &FLAGS_benchmark_filter) ||
+        ParseInt32Flag(argv[i], "benchmark_iterations",
+                       &FLAGS_benchmark_iterations) ||
+        ParseInt32Flag(argv[i], "benchmark_min_iters",
```

Comment:
I wanted to put iterations back so that I didn't remove functionality. I think I will rename benchmark_min_iters and benchmark_max_iters so that their purpose is more clear.